### PR TITLE
test: add unit tests on the ListingView component

### LINKED ISSUE #3217: Core - first unit test on public
**Goal**

Now that we have unit tests in addition to Cypress tests on the public site, there is so much to test on this new smaller scale that we could start anywhere, but we could consider some of the below since they are unit testable and have been bugs in the past. It will probably require mock listing data.

Let's start in the listing detail sidebar in the get application section. It will probably be a test on ListingView.tsx.

**Example test**
The Apply Online button
* If an application is open and it is using the digital application, the Apply Now button should be present
* If it has the digital method but isn't open, the button should not show
* If it has a paper application method and not digital, it should not show

The whole sidebar has a lot of opportunity for something like this (e.g. if pick up address, then show the address, if paper methods, show the download button, etc) but we can start here.

### PR DESCRIPTION
**Pull request by lisunshiny**
# Pull Request Template

## Issue Overview

This PR closes #3217 by adding a unti test on the ListingView component with the recommended tests.

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Mocking all of the appropriate data (e.g. `Jurisidction`, `Listing`) was the primary challenge here. I decided to reuse and extend the existing `testHelpers.ts` data mocks used on the partners side in the public as well by moving that file to the `shared-helpers` folder and importing it from there for both sites.

I also added some stuff that was needed for the tests to work.

## How Can This Be Tested/Reviewed?

Run the new tests, both the new ones and the existing partners tests that I updated (or just check the CI test run which does run the new ListingView test)

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/public/__tests__/components/listing/ListingView.test.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import React from "react"
+import dayjs from "dayjs"
+import { ApplicationMethodType } from "@bloom-housing/backend-core/types"
+import { render, cleanup } from "@testing-library/react"
+import { ListingView } from "../../../src/components/listing/ListingView"
+import { listing, jurisdiction } from "@bloom-housing/shared-helpers/__tests__/testHelpers"
```
Comment (ludtkemorgan) [inline sites/public/__tests__/components/listing/ListingView.test.tsx:6]: 💯

File: sites/public/__tests__/components/listing/ListingView.test.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+afterEach(cleanup)
```
Comment (ludtkemorgan) [inline sites/public/__tests__/components/listing/ListingView.test.tsx:8]: cleanup isn't needed anymore by the [react testing library](https://testing-library.com/docs/react-testing-library/api/#cleanup)

**Review by ludtkemorgan — APPROVED** [review]
Looks good! Thanks for getting this setup!

File: sites/public/__tests__/components/listing/ListingView.test.tsx (hunk 2 above)
Comment (lisunshiny) [reply sites/public/__tests__/components/listing/ListingView.test.tsx]: done

**Review by ColinBuyck — APPROVED** [review]
LGTM 🧊

**Pull request merged**